### ISSUE TITLE
Flowchart

### ISSUE THREAD

**Issue by bryce-carson** [interactivefiction/vimform7#8]
It would be very helpful for maintaining this code base if there was a flowchart for the programs control flow.

**Comment by interactivefiction** [comment]
Is there a specific script you were interested in or is this more of a general request for all scripts?

**Comment by bryce-carson** [comment]
Generally, for all scripts it may be helpful.
I've been working on the open-prj script and I understand the flow very well at this point, but the project directory variable seems inaccessible and remains as the default ERROR when I attempt to access its value.

**Comment by bryce-carson** [comment]
I'll submit a pull request so you can view what I'm encountering.

**Comment by interactivefiction** [comment]
The project directory variable VIMFORM7PROJECTFOLDER is only set when the -f param is provided.  Basically -f sets that variable.  If you want to hack it to be the most recently opened project you might try putting VIMFORM7PROJECTFOLDER=$VIMFORM7LASTOPENED somewhere after the commandline parameters are parsed.  I should also add that there is really no right or wrong way to modify the script.  For example, I really should probably set the project folder to be the last opened at the outset of the script and only modify it if the -f parameter is provided.

**Comment by bryce-carson** [comment]
Yes, I understood that from the time I spent with the script. Strangely, it seemed accessing the variables related to project folders would always give the ERROR value that is established at the beginning of the script, regardless of where in the script I attempted to access it.

I don't have working files anymore to show you that experience, but I have successfully implemented a script to modify the desktop entry for use with gvim and will fork this repo and push my inclusions for review and merging.

It should be POSIX, so there shouldn't be any need to test on major distributions; as long as BASH, sed, cat and echo exist it should work.
